### DISCUSSION TITLE
Alert when field doesn't exist today using @timestamp

### DISCUSSION THREAD

**Discussion by chandankgurram** [jertel/elastalert2#892] [Q&A] [answered]
Hi,
I wanted to send an alert only if the field(body.data.createdOn) does not exists in the today. (body.data.createdOn is in datetime UTC format). I wanted to compare it with the @timestamp. However when I wrote the following rule, 
```
name: "zuora-slack"
type: "any"
index: "zuora"
alert_subject: "Data is not present"
query:
  bool:
    must_not:
      exists:
        field: body.data.createdOn
    filter:
    - range:
        "@timestamp":
          from: now-1d/d
          to: now
alert:
  - "debug"
  - "slack"
slack_webhook_url: "${SLACK_WEBHOOK_URL}"
slack_channel_override: "#my-temp-alerts"` 
```
it didn't work. It just keep sending the alerts, and the recent payload has the field, body.data.createdOn.
So, few things here, 
1) Is my approach correct ?
2) How do we write @timestamp in yaml ? does it work with ""?
3) Do I need to use regex or some sort of pattern match? If so how to do it?

Thanks in advance

**Comment by jertel** [comment, marked as answer]
Can you show me the sample rule, or documentation you based your rule's `query:` block upon? I would expect the top level to have a `filter:` block, and nested within that it would contain the `query:` key, yet yours appears inverted. I'm surprised ElastAlert 2 doesn't complain about that rule file. You may need to use a script filter to handle the "does not exist in the today" part of your requirement. See #819 as an example. I don't know if the `range:` filter can handle the syntax you're using.

**Reply by chandankgurram** [reply]
Hi, I was using this one, 
```GET /_search
{
    "query": {
        "bool": {
            "must_not": {
                "exists": {
                    "field": "user"
                }
            }
        }
    }
}
```
and I made changes as you suggested,
```
filter:
  - query:
      bool:
        must_not:
          exists:
            field: body.data.createdOn
  - script:
      script: doc['body.data.createdOn'].value != [now-1d/d].value 
```
now getting the ERROR:elastalert:Error running query: RequestError(400, 'search_phase_execution_exception', 'compile error')

I restarted the logstash container and recreated the index in kibana again, but no use. 
is this issue due to wrong syntax in the script field?
Or, can we create a new field from the @timestamp field in logstash pipeline, where the new field has only date (DDMMYYYY) and with the new field can we use range filter ?

**Reply by chandankgurram** [reply]
and also I forgot to mention that body.data.createdOn field is always existed from the old logs.

**Reply by jertel** [reply]
Since you're asking for support on how to write a Lucene/ES script I suggest checking with the Elasticsearch project forums. This topic isn't really specific to ElastAlert 2.
